File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/HashInit.java
package com.onevgo.functions;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class HashInit {
    public static MessageDigest hashInit(String algo) {
        try {
            MessageDigest digest = MessageDigest.getInstance(algo);
            return digest;
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static void main(String[] args) {
        MessageDigest md5 = hashInit("md5");
        md5.update("The quick brown fox ".getBytes());
        md5.update("jumped over the lazy dog.".getBytes());
        System.out.println(HashFinal.hashFinal(md5));
    }
}

File: 01_Language/01_Functions/python/hash_init.py
# coding: utf-8

import hashlib


def hash_init(algo):
    return hashlib.new(algo)


if __name__ == '__main__':
    h = hash_init("md5")
    h.update(b"The quick brown fox ")
    h.update(b"jumped over the lazy dog.")
    print(h.hexdigest())
